src/Microsoft.AspNetCore.SystemWebAdapters/HttpFileCollection.cs

```diff
@@ -43,10 +43,14 @@ public override IEnumerator GetEnumerator()
 
     public HttpPostedFile? Get(string name) => FormFiles.GetFile(name) is { } file ? new(file) : null;
 
+    public HttpPostedFile? Get(int index) => FormFiles[index] is { } file ? new(file) : null;
+
     public IList<HttpPostedFile> GetMultiple(string name) => FormFiles.GetFiles(name) is { Count: > 0 } files ? new ReadOnlyPostedFileCollection(files) : Array.Empty<HttpPostedFile>();
 
     public HttpPostedFile? this[string name] => Get(name);
 
+    public HttpPostedFile? this[int index] => Get(index);
```

Comment:
@Tratcher I think this is the first of the namevaluecollection types we can actually fully conform - I hadn't realized IFormFileCollection had a `this[int index]` parameter

Reply:
Ha, yeah, it's a full IReadOnlyList.